### PR TITLE
Initial proposal of the gitlab-ci-workspace docker image

### PR DESCRIPTION
**Pull request by grayside**
This is derived from what I see us using in initial experiments with GitLab CI. I expect it will save some installation time, which is the goal.

While we discussed naming this something like gitlab-ci-runner, that name is commonly used for creating a containerized runner - a host for the kind of container we mean to create, which is more akin to a Jenkins workspace.

You will note that I've seeded this with an initial take on some basic Travis testing for the image.

### CODE REVIEW

**Review by tekante — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,29 @@
+FROM docker:18
```
Comment (febbraro) [inline Dockerfile]: Do we want the version of docker to be a BUILD_ARG?

**Review by febbraro — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (grayside) [reply Dockerfile]: Can we use a build arg in the FROM statement?

If so I am all in favor of adding that parameterization. You'll note that while the travis config here is set up to build the docker image, it's not pushing it, that's simply a comprehensive test at this point.

We could evolve this to supporting multiple versions, but that should be a follow-up.

Comment (febbraro) [reply Dockerfile]: This shows how you can use ARGs in the FROM statement https://docs.docker.com/engine/reference/builder/#understand-how-arg-and-from-interact

Comment (grayside) [reply Dockerfile]: Okay, I've parameterized the Docker version with some inline comments, and added a build.sh script.

**Pull request merged**